data/fields.py

```diff
@@ -17,42 +19,44 @@ def random_string(length=16):
 
 
 class _ResumableSHAField(TextField):
+    """
+    Base Class used to store the state of an in-progress hash in the database. This is particularly
+    useful for working with large byte streams and allows the hashing to be paused and resumed
+    as needed.
+    """
+
     def _create_sha(self):
         raise NotImplementedError
 
     def db_value(self, value):
+        """
+        Serialize the Hasher's state for storage in the database as plain-text.
+        """
         if value is None:
             return None
 
-        sha_state = value.state()
-
-        # One of the fields is a byte string, let's base64 encode it to make sure
-        # we can store and fetch it regardless of default collocation.
-        sha_state[3] = base64.b64encode(sha_state[3])
-
-        return json.dumps(sha_state)
+        serialized_state = base64.b64encode(pickle.dumps(value)).decode("ascii")
```

Comment:
Please link this line in the JIRA to file: pickling will likely not work long-term, and there are back-compat concerns

Reply:
https://issues.redhat.com/browse/PROJQUAY-771